### DISCUSSION TITLE
How do I change what is included in the result of Language.evaluate?

### DISCUSSION THREAD

**Discussion by graue70** [explosion/spaCy#7427] [Help: Coding & Implementations] [answered]
I have some default components and a custom component which defines a `score()` method calculating the score for 'my_number'. If I run `nlp.evaluate(examples)`, I get a dict containing 14 values, only one of which is my 'my_number'.

**How can I exclude values from the result?**

I tried using `default_score_weights={"my_number": None}` but it seems to have no effect. I also wouldn't be able to use this for default components as I don't control their factories.

*This is example output:*
```
{'token_acc': 1.0, 'token_p': 1.0, 'token_r': 1.0, 'token_f': 1.0, 'tag_acc': None, 'ents_p': None, 'ents_r': None, 'ents_f': None, 'ents_per_type': None, 'morph_acc': None, 'morph_per_feat': None, 'lemma_acc': None, 'pos_acc': None, 'my_number': 0.75}
```

*I want something like this instead:*
```
{'token_f': 1.0,'ents_f': None, 'my_number': 0.75}
```

**Comment by polm** [comment, marked as answer]
Since it's just a dict you can filter the keys like any other dict.

```
raw_dict = nlp.evaluate(examples)
keep_keys = ["token_f", "ents_f", "my_number"]
clean_dict = {k:v for k, v in raw_dict.items() if k in keep_keys}
```

Is that what you needed, or was there some other reason the other keys were an issue?

**Reply by graue70** [reply]
Since there is `default_score_weights` which from my understanding specifies what scores are relevant and how relevant (when training), I thought there would be something like that for the evaluation as well.

I guess since it wouldn't change anything except the output, there is no disadvantage to using your solution.
